3.2/root/usr/share/container-scripts/mongodb/init-petset-replset.sh

```diff
@@ -0,0 +1,185 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source "${CONTAINER_SCRIPTS_PATH}/common.sh"
+
+# This is a full hostname that will be added to replica set
+# (for example, "replica-2.mongodb.myproject.svc.cluster.local")
+readonly MEMBER_HOST="$(hostname -f)"
+
+# Description of possible statuses: https://docs.mongodb.com/manual/reference/replica-states/
+readonly WAIT_PRIMARY_STATUS='
+  while (rs.status().startupStatus || (rs.status().hasOwnProperty("myState") && rs.status().myState != 1)) {
+    printjson(rs.status());
+    sleep(1000);
+  };
+  printjson(rs.status());
+'
+readonly WAIT_SECONDARY_STATUS="
+  var mbrs;
+  while (!mbrs || mbrs.length == 0 || mbrs[0].state != 2) {
+    printjson(rs.status());
+    sleep(1000);
+    mbrs = rs.status().members.filter(function(el) {
+      return el.name.indexOf(\"${MEMBER_HOST}:\") > -1;
+    });
+  };
+  print(mbrs[0].stateStr);
+"
+readonly FIND_PRIMARY_HOST='
+  var name = rs.status().members.filter(function(el) {
+    return el.stateStr == "PRIMARY";
+  })[0].name;
+  print("PRIMARY|" + name);
+'
+readonly ADD_MEMBER="
+  var members = rs.status().members.filter(function(el) {
+    return el.name.indexOf(\"${MEMBER_HOST}:\") > -1;
+  });
+  if (members.length == 0) {
+    printjson(rs.add('${MEMBER_HOST}'));
+  } else {
+    print('{ \"ok\" : 1 }');
+  }
+"
+
+# Outputs available endpoints (hostnames) to stdout.
+# This also includes hostname of the current pod.
+#
+# Uses the following global variables:
+# - MONGODB_SERVICE_NAME (optional, defaults to 'mongodb')
+function find_endpoints() {
+    local service_name="${MONGODB_SERVICE_NAME:-mongodb}"
+
+    # Extract host names from lines like this: "10 33 0 mongodb-2.mongodb.myproject.svc.cluster.local."
+    dig "${service_name}" SRV +search +short | cut -d' ' -f4 | rev | cut -c2- | rev
+}
+
+# Performs initialization of the primary node.
+#
+# Arguments:
+# - $1 hostname
+#
+# Uses the following global variables:
+# - MONGODB_REPLICA_NAME
+# - MONGODB_ADMIN_PASSWORD
+# - WAIT_PRIMARY_STATUS
+function setup_primary() {
+    local member_host="${1}"
+
+    local endpoints
+    endpoints="$(find_endpoints)"
+
+    if [ "$(echo "${endpoints}" | grep -v "^${member_host}" | wc -l)" != '0' ]; then
```

Comment:
When PetSet controller creates pods it creates them one-by-one. If we're looking for other members and they're already there, then it means that the first pod were restarted. In this case, we assume that we've initialized replica set before and we're already in this replica set.